### PR TITLE
IVC/Poseidon: make the round constants public values

### PR DESCRIPTION
**Pull request by dannywillems**
Built on top of !https://github.com/o1-labs/proof-systems/pull/2306
in 2306, the round constants are considered as witness values. Simply making them public to not fold them later.

### CODE REVIEW

**Comment by dannywillems** [comment]
Investigating CI, seems to be related to the IVC circuit. https://github.com/o1-labs/proof-systems/actions/runs/9505793514/job/26201502174?pr=2314

**Comment by dannywillems** [comment]
Related to the fixed selectors. Must be mapped correctly in the IVC circuit.

File: ivc/src/ivc/columns.rs [hunk 1]
```diff
@@ -25,6 +25,9 @@ pub fn block_height<const N_COL_TOTAL: usize, const N_CHALS: usize>(block_num: u
 pub const IVC_POSEIDON_STATE_SIZE: usize = 3;
 pub const IVC_POSEIDON_NB_FULL_ROUND: usize = 55;
 
+pub const IVC_NB_TOTAL_FIXED_SELECTORS: usize =
+    IVC_POSEIDON_NB_FULL_ROUND * IVC_POSEIDON_STATE_SIZE + N_BLOCKS;
```
Comment (volhovm) [inline ivc/src/ivc/columns.rs:29]: Cosmetic: maybe create  `N_FSEL_POSEIDON = IVC_POSEIDON_NB_FULL_ROUND * IVC_POSEIDON_STATE_SIZE` and refer to that?

Comment (volhovm) [reply ivc/src/ivc/columns.rs:29]: Generally where possible it's good to keep to the same notation `prover` is using: `N_WIT, N_FSEL, N_DSEL, N_REL` etc

File: ivc/src/ivc/interpreter.rs [hunk 2]
```diff
@@ -887,16 +889,23 @@ where
 }
 
 /// Builds selectors for the IVC circuit.
+/// The round constants for Poseidon are not added in this function, and must be
+/// done separately.
+/// The size of the array is the total number of public values required for the
+/// IVC. Therefore, it includes the potential round constants required by
+/// the hash function.
+// FIXME: rc should be handled here or in the lens
```
Comment (volhovm) [inline ivc/src/ivc/interpreter.rs:897]: I'm working on this (slowly, there's a PR...) -- my current idea is to put it into `ColumnIndexer`, and thus the `fixed_selectors` for `IVCColumn` would call `fixed_selectors` for `PoseidonColumn`... For now this is fine, but I'd also prefer to just add `build_selectors` function for poseidon, and call it from this IVC function. Anyway, if it works it works :)

File: ivc/src/ivc/mod.rs [hunk 3]
```diff
@@ -91,8 +94,20 @@ mod tests {
...
 
         println!("Building fixed selectors");
-        let fixed_selectors: Vec<Vec<Fp>> =
-            build_selectors::<_, TEST_N_COL_TOTAL, TEST_N_CHALS>(domain_size).to_vec();
+        let mut fixed_selectors: Vec<Vec<Fp>> =
+            build_selectors::<Fp, TEST_N_COL_TOTAL, TEST_N_CHALS>(domain_size).to_vec();
+
+        // Write constants
+        {
+            let rc = PoseidonBN254Parameters.constants();
```
Comment (volhovm) [inline ivc/src/ivc/mod.rs:102]: Any reason this block that's repeated twice cannot be a separate function `build_selectors() -> Vec<Vec<Fp>>`?

**Review by volhovm — APPROVED** [review]
Cool! One cosmetic comment: the "building fixed selectors" block of 15 lines is repeated exactly 8 times, so unless it's problematic for some reason it'd be nice to have it in a separate function. Or two. I realise different Poseidons have different constants...

File: ivc/src/ivc/mod.rs (hunk 3 above)
Comment (dannywillems) [reply ivc/src/ivc/mod.rs:102]: Interpreter depends on the field.

**Pull request merged**